ops/framework.py

```diff
@@ -1104,8 +1104,25 @@ def __init__(self):
         self.parent_type = None  # type: Optional[Type[Any]]
         self.attr_name = None  # type: Optional[str]
 
-    def __get__(self, parent: '_ObjectType', parent_type: 'Type[_ObjectType]'
-                ) -> Union['StoredState', BoundStoredState]:
+    if TYPE_CHECKING:
+        @typing.overload
+        def __get__(
+                self,
+                parent: Literal[None],
+                parent_type: 'Type[_ObjectType]') -> 'StoredState':
+            pass
+
+        @typing.overload
+        def __get__(
+                self,
+                parent: '_ObjectType',
+                parent_type: 'Type[_ObjectType]') -> BoundStoredState:
+            pass
```

Comment:
is it necessary to have an overload only in typechecker (this of course will require import modifications)?
also, pass could be replaced by `...` for the body of the function
```suggestion
    @typing.overload
    def __get__(
            self,
            parent: Literal[None],
            parent_type: 'Type[_ObjectType]') -> 'StoredState':
        ...

    @typing.overload
    def __get__(
            self,
            parent: '_ObjectType',
            parent_type: 'Type[_ObjectType]') -> BoundStoredState:
        ...
 
```